Unity-Technologies/ScriptableRenderPipeline
# Fix custom pass depth copy

### Purpose of this PR
Fix the depth buffer copy that happend before the custom pass injection point 'After Opaque And Normal' when decals were enabled.
https://fogbugz.unity3d.com/f/cases/1229891/

This bug caused the modification of the depth buffer to notr be taken in account in the depth pyramid, resulting in visible artifacts when using effect that needs the depth pyramid (SSAO for example).

**Manual Tests**: What did you do?

My test scene consist in two custom passes:
- one in after opaque and normal mode that renders the cube in front, it renders the cube in the depth buffer and then in the normal buffer to ensure all the data needed for screen space effects is correct.
- and the other custom pass renders the same object but with the Forward pass to make it visible.

Before the fix a such setup with SSAO did that:

![BeforeFIx](https://user-images.githubusercontent.com/6877923/78056483-7e98a900-7385-11ea-8405-18b9837d778a.gif)
As you can see, the SSAO leaks in front of the object because the depth of the cube is not in the depth pyramid.

This is after the fix:

![AfterFix](https://user-images.githubusercontent.com/6877923/78056490-822c3000-7385-11ea-8a0e-93f92bc3418c.gif)
